### PR TITLE
Fix aave

### PR DESCRIPTION
**Pull request by 0xpeluche**

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/aave exports TVL: 
        
 
 
    ethereum-staking          458.64 M
    staking                   458.64 M
    polygon                   58.75 M
    avax                      23.97 M
    borrowed                  19.68 M
    polygon-borrowed          15.04 M
    avax-borrowed             4.64 M
    ethereum-pool2            3.56 M
    pool2                     3.56 M
    ethereum                  0
    ethereum-borrowed         0
    
    total                    82.72 M

**Comment by llamatester** [comment]
The adapter at projects/aave-v3 exports TVL: 
        
 
 
    ethereum                  17.74 B
    borrowed                  8.66 B
    ethereum-borrowed         7.28 B
    arbitrum                  1.25 B
    avax                      569.44 M
    arbitrum-borrowed         556.74 M
    polygon                   390.36 M
    scroll                    372.75 M
    optimism                  232.01 M
    base                      223.12 M
    avax-borrowed             216.98 M
    scroll-borrowed           200.93 M
    polygon-borrowed          138.24 M
    base-borrowed             114.70 M
    bsc                       96.70 M
    optimism-borrowed         90.66 M
    xdai                      75.55 M
    metis                     35.14 M
    bsc-borrowed              28.02 M
    xdai-borrowed             22.39 M
    metis-borrowed            15.13 M
    era                       4.17 M
    harmony-borrowed          999.97 k
    harmony                   946.38 k
    era-borrowed              704.32 k
    fantom                    60.35 k
    fantom-borrowed           12.50 k
    
    total                    20.98 B

**Comment by g1nt0ki** [comment]
hi @0xpeluche is this ready for review?

**Comment by 0xpeluche** [comment]
Oh yeah sorry, forgot to remove WIP key

File: projects/aave-v3/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+const borrowed = async (api, poolData) => {
+  await fetchReserveData(api, poolData, 'totalVariableDebt');
+};
+
+Object.keys(CONFIG).forEach((chain) => {
+  const poolDatas = CONFIG[chain];
+  module.exports[chain] = {
+    methodology: "Counts the tokens locked in the contracts to be used as collateral to borrow or to earn yield. Borrowed coins are not counted towards the TVL, so only the coins actually locked in the contracts are counted. There's multiple reasons behind this but one of the main ones is to avoid inflating the TVL through cycled lending.",
```
Comment (g1nt0ki) [inline projects/aave-v3/index.js:57]: for tvl adapters, methodology is common/should be on the top level

File: projects/aave-v3/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  tokensWithPools.forEach(({ tokens }, poolIndex) => {
+    tokens.forEach((token, i) => {
+      api.add(token, reserveBalances[poolIndex][i][dataKey]);
+    });
+  });
+};
+
+const tvl = async (api, poolData) => {
+  await fetchReserveData(api, poolData, 'totalAToken');
```
Comment (g1nt0ki) [inline projects/aave-v3/index.js:47]: this would return value including the amount lent out, one of the reasons why it is safe to check token balance

**Comment by llamatester** [comment]
The adapter at projects/aave-v3 exports TVL: 
        
 
 
    ethereum                  10.30 B
    borrowed                  8.56 B
    ethereum-borrowed         7.15 B
    arbitrum                  669.41 M
    arbitrum-borrowed         544.50 M
    avax                      348.14 M
    avax-borrowed             279.09 M
    polygon                   246.80 M
    scroll-borrowed           179.38 M
    scroll                    155.74 M
    optimism                  138.90 M
    polygon-borrowed          136.64 M
    base                      119.04 M
    base-borrowed             115.64 M
    optimism-borrowed         89.83 M
    bsc                       66.86 M
    xdai                      52.76 M
    bsc-borrowed              28.02 M
    xdai-borrowed             22.17 M
    metis                     19.49 M
    metis-borrowed            14.98 M
    era                       3.37 M
    harmony-borrowed          976.23 k
    era-borrowed              738.73 k
    fantom                    61.62 k
    fantom-borrowed           11.98 k
    harmony                   0
    
    total                    12.12 B

**Pull request merged**